bleak/__init__.py

```diff
@@ -8,13 +8,39 @@
 import os
 import sys
 import logging
-import platform
 import asyncio
 
 from bleak.__version__ import __version__  # noqa: F401
 from bleak.exc import BleakError
-
-_on_rtd = os.environ.get("READTHEDOCS") == "True"
+from bleak._api import (
+    BleakScanner,
+    _BleakScannerImplementation,
+    BleakClient,
+    _BleakClientImplementation,
+    BLEDevice,
+    _BLEDeviceImplementation,
+    BleakGATTServiceCollection,
+    BleakGATTService,
+    _BleakGATTServiceImplementation,
+    BleakGATTCharacteristic,
+    _BleakGATTCharacteristicImplementation,
+    BleakGATTDescriptor,
+    _BleakGATTDescriptorImplementation,
+    discover,
+)
+
+__all__ = [
+    "BleakError",
+    "BleakScanner",
+    "BleakClient",
+    "BLEDevice",
+    "BleakGATTServiceCollection",
+    "BleakGATTService",
+    "BleakGATTCharacteristic",
+    "BleakGATTDescriptor",
+    "discover",
+    "cli",
+]
```

Comment:
I think we also need `AdvertisementData`, `AdvertisementDataCallback` and `AdvertisementDataFilter` (from `backends.scanner`) here.

Reply:
And perhaps other types reference via type hints of the top-level APIs?

Reply:
Will do for the `backends.scanner` objects. But I'm not sure I understand what you mean by 

> And perhaps other types reference via type hints of the top-level APIs?

can you explain?

Reply:
Hmm, I'm also getting a funny signature for `AdvertisementDataCallback`, and `AdvertisementDataFilter`, and they aren't the same type as what is used in the `BleakScanner` signatures... Any suggestion why that could be the case?

Reply:
I need help with these ones. I _think_ `TypeAlias` might help, or `NewType`, but I'm not familiar with them at all, and I also think they're not backward compatible to 3.7.

Reply:
> can you explain?

I didn't go through and check every method to see if there are other custom types like these. We will need to do this eventually to make sure we didn't miss anything.

Reply:
> Hmm, I'm also getting a funny signature

Do you mean in the docs generated with Sphinx? We are using such and old version of Sphinx, we really should update.